src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -799,6 +799,25 @@ private static bool CanReplace(ISymbol symbol)
                 }
             }
 
+            // Try to eliminate cases without actually calling CanReplaceWithReducedName. For expressions of the form
+            // 'this.Name' or 'base.Name', no additional check here is required.
+            if (!memberAccess.Expression.IsKind(SyntaxKind.ThisExpression, SyntaxKind.BaseExpression))
+            {
+                var actualSymbol = semanticModel.GetSymbolInfo(memberAccess.Name, cancellationToken);
+                if (actualSymbol.GetAnySymbol() is null)
+                {
+                    return false;
+                }
+
+                // Get a syntax node that's not part of the tree to make sure we speculate on just the name.
+                var speculativeTree = memberAccess.Name.WithoutTrivia();
```

Comment:
Cool, I hadn't used `LookupSymbols` before. It turns out that the complete check needs to use `LookupSymbols` as an initial check (the fastest expected path), and `LookupNamespacesOrTypes` to cover the special case seen in `SimplifyTypeNamesTests.TestColorColorCase1`. This update removed the need to pull `memberAccess.Name` out of the tree and also fixed the previously failing test cases.